ingestion/functions/common/parsing_lib.py

```diff
@@ -154,13 +157,28 @@ def write_to_server(
         # End of batch.
         if not batch:
             break
+        # There are two exponential backoffs:
+        # * wait, total_wait keeps track of the backoff from 5xx errors
+        # * conn_wait, total_conn_wait tracks backoff for connection errors
         total_wait = 0
+        total_conn_wait = 0
         wait = 10  # initial wait time in seconds
+        conn_wait = 30
         print(f"Sending {len(batch)} cases, total so far: {counter['total']}")
         # Exponential backoff in dev and prod, but not for local testing
-        while total_wait <= (MAX_WAIT_TIME if env in ["dev", "prod"] else 0):
-            res = requests.post(put_api_url, json={"cases": batch},
-                                headers=headers, cookies=cookies)
+        while (
+            total_wait <= (MAX_WAIT_TIME if env in ["dev", "prod"] else 0)
+            and total_conn_wait < MAX_CONN_WAIT_TIME
+        ):
+            try:
+                res = requests.post(put_api_url, json={"cases": batch},
+                                    headers=headers, cookies=cookies)
+            except requests.exceptions.ConnectionError:
+                print(f"Failed to connect to data service, waiting {conn_wait}s")
+                time.sleep(conn_wait)
```

Comment:
Could we use the `timeout` parameter of `.post()` rather than `time.sleep()`?

Reply:
Didn't know about that! With no timeout, requests will wait for the data service which I think is the correct approach when inserting cases as we don't know how long that'll take.